core/api/src/app/wallets/index.ts

```diff
@@ -16,14 +16,35 @@ export * from "./settle-payout-txn"
 export * from "./update-legacy-on-chain-receipt"
 export * from "./update-pending-invoices"
 export * from "./validate"
+export * from "./get-invoice-for-wallet-by-hash"
 
+import { CouldNotFindWalletFromIdError } from "@/domain/errors"
 import { WalletsRepository } from "@/services/mongoose"
 
 export const getWallet = async (walletId: WalletId) => {
   const wallets = WalletsRepository()
   return wallets.findById(walletId)
 }
 
+export const getWalletForAccountById = async ({
```

Comment:
is it an authorization related function? maybe should be named as such if that the intention of this function

Reply:
What would you change the name to? The function will look for a wallet that belongs to an account. If the wallet exists but does not belong to that account, it will return a CouldNotFindWalletError.

Reply:
maybe `checkWalletForAccountById`?

Reply:
Not sure if that works here. It's not a check that returns true or false, it either returns a wallet or a CouldNotFindWalletError.